diff --git a/fundamentals/src/asset.rs b/fundamentals/src/asset.rs
index 120f3a8..a49a7ab 100644
--- a/fundamentals/src/asset.rs
+++ b/fundamentals/src/asset.rs
@@ -44,31 +44,31 @@ const DOT_DECIMALS: u32 = 10;
 // Construct these assets from the perspective of AssetHub (1000).
 parameter_types! {
 	// `Assets` instance that contains no assets.
-	pub EmptyAssets: Assets = todo!();
+	pub EmptyAssets: Assets = vec![].into();
 	// USDT.
-	pub Usdt: AssetId = todo!();
+	pub Usdt: AssetId = (PalletInstance(50), GeneralIndex(1984)).into();
 	// The native token of the relay chain, i.e. DOT.
-	pub DotToken: AssetId = todo!();
+	pub DotToken: AssetId = Parent.into();
 	// 100 of the parachain's native token
-	pub OneHundredUsdt: Asset = todo!();
+	pub OneHundredUsdt: Asset = (Usdt::get(), 100u128 * 10u128.pow(HDX_DECIMALS)).into();
 	// Some amount of the native token of the relay chain.
-	pub OneHundredDot: Asset = todo!();
+	pub OneHundredDot: Asset = (DotToken::get(), 100u128 * 10u128.pow(DOT_DECIMALS)).into();
 }
 
 // Non-Fungible Tokens
 parameter_types! {
 	// Location of NFT collection with id 42 inside of the uniques pallet in Polkadot parachain 1000.
-	pub NftLocation: Location = todo!();
+	pub NftLocation: Location = Location::new(1, [Parachain(1000), PalletInstance(52), GeneralIndex(3)]);
 	// The NFT with id 69 inside of that collection.
-	pub Nft: Asset = todo!();
+	pub Nft: Asset = (NftLocation::get(), 69u64).into();
 }
 
 // Asset Filters
 parameter_types! {
 	// A filter which will capture all possible assets.
-	pub AllAssetsFilter: AssetFilter = todo!();
+	pub AllAssetsFilter: AssetFilter = AssetFilter::Wild(WildAsset::All);
 	// A filter specific for the DOT Token
-	pub DotFilter: AssetFilter = todo!();
+	pub DotFilter: AssetFilter = OneHundredDot::get().into();
 	// A filter specific for the Native Token
-	pub UsdtFilter: AssetFilter = todo!();
+	pub UsdtFilter: AssetFilter = OneHundredUsdt::get().into();
 }
