diff --git a/fundamentals/src/asset.rs b/fundamentals/src/asset.rs
index 8b13789..120f3a8 100644
--- a/fundamentals/src/asset.rs
+++ b/fundamentals/src/asset.rs
@@ -1 +1,74 @@
+//! # Fundamentals Lesson 2
+//!
+//! All locations in this module are relative to Polkadot parachain 2000.
 
+use frame_support::parameter_types;
+use xcm::latest::prelude::*;
+
+// Polkadot Topography
+//
+//                              ┌───────────┐
+//                              │  Relay A  │
+//                              │  Polkadot │
+//                              └─────┬─────┘
+//                                    │
+//                         ┌──────────┴──────────┐
+//                         │                     │
+//                   ┌─────┴─────┐         ┌─────┴─────┐
+//                   │  AssetHub │         │  HydraDx  │
+//                   │  Id 1000  │         │  Id 2034  │
+//                   └─────┬─────┘         └───────────┘
+//                         │
+//                  ┌──────┴──────┐
+//                  │             │
+//           ┌──────┴─────┐ ┌─────┴──────┐
+//           │   Pallet   │ │   Pallet   │
+//           │   Assets   │ │    NFT     │
+//           │            │ │            │
+//           │ Pallet #50 │ │ Pallet #52 │
+//           └─────┬──────┘ └─────┬──────┘
+//                 │              │
+//       ┌─────────┴───┐          └────┬──────────────┐
+//       │             │               │              │
+// ┌─────┴─────┐ ┌─────┴─────┐  ┌──────┴─────┐ ┌──────┴─────┐
+// │   Asset   │ │   Asset   │  │ Collection │ │ Collection │
+// │   USDC    │ │   USDT    │  │   Kitties  │ │  Zombies   │
+// │           │ │           │  │            │ │            │
+// │ Id 1337   │ │  Id 1984  │  │    Id 3    │ │    Id 66   │
+// └───────────┘ └───────────┘  └────────────┘ └────────────┘
+
+const HDX_DECIMALS: u32 = 12;
+const DOT_DECIMALS: u32 = 10;
+
+// Fungible Tokens
+// Construct these assets from the perspective of AssetHub (1000).
+parameter_types! {
+	// `Assets` instance that contains no assets.
+	pub EmptyAssets: Assets = todo!();
+	// USDT.
+	pub Usdt: AssetId = todo!();
+	// The native token of the relay chain, i.e. DOT.
+	pub DotToken: AssetId = todo!();
+	// 100 of the parachain's native token
+	pub OneHundredUsdt: Asset = todo!();
+	// Some amount of the native token of the relay chain.
+	pub OneHundredDot: Asset = todo!();
+}
+
+// Non-Fungible Tokens
+parameter_types! {
+	// Location of NFT collection with id 42 inside of the uniques pallet in Polkadot parachain 1000.
+	pub NftLocation: Location = todo!();
+	// The NFT with id 69 inside of that collection.
+	pub Nft: Asset = todo!();
+}
+
+// Asset Filters
+parameter_types! {
+	// A filter which will capture all possible assets.
+	pub AllAssetsFilter: AssetFilter = todo!();
+	// A filter specific for the DOT Token
+	pub DotFilter: AssetFilter = todo!();
+	// A filter specific for the Native Token
+	pub UsdtFilter: AssetFilter = todo!();
+}
