diff --git a/fundamentals/src/pallet_xcm.rs b/fundamentals/src/pallet_xcm.rs
index c3ea023..baec7b7 100644
--- a/fundamentals/src/pallet_xcm.rs
+++ b/fundamentals/src/pallet_xcm.rs
@@ -124,10 +124,9 @@ pub mod pallet {
 impl<T: Config> Pallet<T> {
 	/// Execute an XCM locally on this chain on behalf of `origin`.
 	pub fn do_execute(origin: OriginFor<T>, message: Xcm<()>) -> DispatchResult {
-		// Use `ExecuteXcmOrigin` to "ensure" that `origin` is able to execute a local XCM
-		// and assign the resulting location to `execute_origin`.
-		// Use `XcmExecutor` to `execute` the call, and if an error occurs, return `ExecutorError`.
-		todo!("{:?}", message)
+		let execute_origin: Location = T::ExecuteXcmOrigin::ensure_origin(origin)?;
+		T::XcmExecutor::execute(execute_origin, message).map_err(|_| Error::<T>::ExecutorError)?;
+		Ok(())
 	}
 
 	/// Relay an XCM `message` from a given `interior` location in this context to a given `dest`
